### DISCUSSION TITLE
完形填空（Cloze）

### DISCUSSION THREAD

**Discussion by tangshimin** [tangshimin/typing-learner#14] [Ideas]
我现在想到的交互流程如下：
1. 导入文本
2. 选择挖空模式（多选）
    -  随机挖空
    - 手动选择，快捷键 Ctrl + B， 鼠标选择文本，然后确定。
    - 选择一个生词本，自动挖空。
    - 选择词性，然后自动挖空。
3.  然后呢？让用户输入挖空的单词？用 `Tab` 切换，全部输入完了判断对错，还是一个单词一个单词的判断对错？

**Comment by tangshimin** [comment]
可以在学习完一个章节后的对话框中加入一个  ⌈填空⌋ 选项
![Snipaste 20](https://user-images.githubusercontent.com/16540656/170868398-ca7361f4-5bc2-4e7e-a3ca-f54c86dfde83.png)

这个选项只出现在使用 MKV 或字幕生成的词库中，用户选择  ⌈填空⌋ 后，就像默写单词一样，一次出现一句 ⌈挖空⌋ 后字幕，用户输入被挖的单词或者从四个选项中选择正确的答案。
挖空模式有两种，一种是挖刚刚学过的单词，另一种是随机挖一个单词。

**Reply by yzh979** [reply]
这个完形填空功能是我在网上找了一年多，非常想要的一个功能。希望您能开发出来啊。太好啦！谢谢您啊，加油

**Reply by tangshimin** [reply]
> 这个完形填空功能是我在网上找了一年多，非常想要的一个功能。希望您能开发出来啊。太好啦！谢谢您啊，加油

你想挖字幕的的生词？还是随机挖一个？

**Reply by tangshimin** [reply]
> 可以在学习完一个章节后的对话框中加入一个 ⌈填空⌋ 选项 ![Snipaste 20](https://user-images.githubusercontent.com/16540656/170868398-ca7361f4-5bc2-4e7e-a3ca-f54c86dfde83.png)
> 
> 这个选项只出现在使用 MKV 或字幕生成的词库中，用户选择 ⌈填空⌋ 后，就像默写单词一样，一次出现一句 ⌈挖空⌋ 后字幕，用户输入被挖的单词或者从四个选项中选择正确的答案。 挖空模式有两种，一种是挖刚刚学过的单词，另一种是随机挖一个单词。

想把交互改成，先播放字幕，然后再让用户输入被挖的生词。

**Reply by yzh979** [reply]
挖空生词和随机挖空一个单词，应该都有需求啊。
